cypher/cypher-docs/src/docs/dev/query-tuning.asciidoc

```diff
@@ -125,6 +126,95 @@ In Enterprise Edition, the Cypher query planner selects the runtime, falling bac
+=== Cypher planner
+The Cypher planner takes a Cypher query and computes an execution plan that solves it.
+For any given query there is likely a number of execution plan candidates that each solve the query in a different way.
+The planner uses a search algorithm to find the execution plan with the lowest estimated execution cost.
+
+This table describes the available planner options:
+[options="header"]
+|===
+| Query option | Description | Default
+| `planner=cost` | Use cost based planning with default limits on plan search space and time. | X
+| `planner=idp`  | Synonym for `planner=cost`. |
+| `planner=dp`
+| Use cost based planning without limits on plan search space and time to perform an exhaustive search for the best execution plan.
+  Note that using this option may significantly increase the planning time of the query. |
+|===
+
+[[cypher-connect-components-planner]]
+=== Cypher connect-components planner
+One part of the Cypher planner is responsible for combining sub-plans for separate patterns into larger plans - a task referred to as connecting components.
+
+This table describes the available query options for the connect-components planner:
+[options="header"]
+|===
+| Query option | Description | Default
+| `connectComponentsPlanner=greedy` | Use a greedy approach when combining sub-plans. | X
+| `connectComponentsPlanner=idp`
+| Use the cost based IDP search algorithm when combining sub-plans.
+  Note that using this option may significantly increase the planning time of the query. |
+|===
+
+[[cypher-update-strategy]]
+=== Cypher update strategy
+This option affects eagerness of updating queries
+[options="header"]
+|===
+| Query option | Description | Default
+| `updateStrategy=default` | Update queries are executed eagerly when needed. | X
+| `updateStrategy=eager`   | Update queries are always executed eagerly. |
+|===
+
+[[cypher-expression-engine]]
+=== Cypher expression engine
+This option affects how the runtime evaluates expressions. The possible values are:
+[options="header"]
+|===
+| Query option | Description | Default
+| `expressionEngine=default`     | Compile expressions and use the compiled expression engine when needed. | X
+| `expressionEngine=interpreted` | Always use the interpreted expression engine. |
+| `expressionEngine=compiled`    | Always compile expressions and use the compiled expression engine. Cannot be used together with `runtime=interpreted` |
+|===
+
+[[cypher-operator-engine]]
+=== Cypher operator engine
+This query option affects whether the pipelined runtime attempts to fuse groups of operators into pipelines. The possible values are:
+[options="header"]
+|===
+| Query option | Description | Default
+| `operatorEngine=default`     | Attempt to fuse operators when applicable. | X
+| `operatorEngine=interpreted` | Never attempt to fuse operators. |
+| `operatorEngine=compiled`    | Always attempt to fuse operators. Cannot be used together with `runtime=interpreted` or `runtime=slotted` |
```

Comment:
"fuse" might sound misleading when the option name is "compiled".

Maybe this is enough?:
```suggestion
This query option affects whether the pipelined runtime attempts to generate compiled code for groups of operators. The possible values are:
[options="header"]
|===
| Query option | Description | Default
| `operatorEngine=default`     | Attempt to generate compiled operators when applicable. | X
| `operatorEngine=interpreted` | Never attempt to generate compiled operators. |
| `operatorEngine=compiled`    | Always attempt to generate compiled operators. Cannot be used together with `runtime=interpreted` or `runtime=slotted` |
```